src/components/ProjectSettingsTabs/GeneralTab/Fields/RepoConfigField.tsx

```diff
@@ -0,0 +1,251 @@
+import { useState } from "react";
+import { useMutation } from "@apollo/client";
+import styled from "@emotion/styled";
+import { Field } from "@rjsf/core";
+import { Button } from "components/Button";
+import { ConfirmationModal } from "components/ConfirmationModal";
+import { SpruceForm } from "components/SpruceForm";
+import { size } from "constants/tokens";
+import { useToastContext } from "context/toast";
+import {
+  AttachProjectToRepoMutation,
+  AttachProjectToRepoMutationVariables,
+  DetachProjectFromRepoMutation,
+  DetachProjectFromRepoMutationVariables,
+} from "gql/generated/types";
+import {
+  ATTACH_PROJECT_TO_REPO,
+  DETACH_PROJECT_FROM_REPO,
+} from "gql/mutations";
+
+interface ModalProps {
+  onCancel: () => void;
+  onConfirm: (formUpdate: any) => void;
+  open: boolean;
+}
+
+export const MoveRepoModal: React.FC<ModalProps> = ({
+  onCancel,
+  onConfirm,
+  open,
+}) => {
+  const [formState, setFormState] = useState({});
+  const [hasError, setHasError] = useState(false);
+  const isDisabled =
+    Object.keys(formState).length === 0 ||
+    Object.values(formState).some((input) => input === "") ||
+    hasError;
```

Comment:
I wonder can we not rely on rjsf form validation instead of validating it outside?